packages/server/src/index.ts

```diff
@@ -101,15 +96,26 @@ export const start = async (
     stopPort: port + portTriesLimit,
   });
 
-  if (actualPort !== port) {
-    process.stdout.write(
-      `Port ${port} is already taken. Starting server on port ${actualPort} instead.\n`,
-    );
-  }
+  let started = false;
+  bundler.on("bundled", () => {
+    if (!started) {
+      started = true;
 
-  server.listen(actualPort, () => {
-    process.stdout.write(
-      `Server is running at http://localhost:${actualPort}.\n`,
-    );
+      process.stdout.write(`Starting server on port ${port}...\n`);
+
+      if (actualPort !== port) {
+        process.stdout.write(
+          `Port ${port} is already taken. Starting server on port ${actualPort} instead.\n`,
+        );
+      }
+
+      server.listen(actualPort, () => {
+        process.stdout.write(
+          `Server is running at http://localhost:${actualPort}.\n`,
+        );
+      });
+    }
   });
```

Comment:
The other way to do it might be to do something like:

```javascript
bundler.on("bundled", startServer);

const startServer = () => {
  // This is the important line.
  bundler.off("bundled", startServer);

  process.stdout.write(`Starting server on port ${port}...\n`);
  
  if (actualPort !== port) {
    process.stdout.write(
      `Port ${port} is already taken. Starting server on port ${actualPort} instead.\n`,
    );
  }
  
  server.listen(actualPort, () => {
    process.stdout.write(
      `Server is running at http://localhost:${actualPort}.\n`,
    );
  });
};
```

Reply:
Or even:

```javascript
new Promise((resolve, reject) => {
  bundler.on("bundled", resolve);
  bundler.on("error", reject);
}).then(startServer);
```